### PR TITLE
Start and end a packet

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds about the simplest copy of packet start/end/metadata write I can think of, while supporting all the keys bits we'll need. It could have been broken up a little as there are two things done here in addition:

* trivial Root object which holds all the bits that we need to push around at once
* update the schemas to 0.1.1

The actual packet workflow is most easily seen in tests where we start and end a packet. At present all files are included, later we'll expand to shadow the orderly2 options of immutable and ignored files.  The implementation actually follows very closely to the R version, with a mutable packet object that we'll pass about and update

### CODE REVIEW

File: scripts/update_schemas [hunk 1]
```diff
@@ -23,6 +23,5 @@ writeLines(c("# Imported from outpack",
              "Do not make changes to files here, they will be overwritten",
              "Run ./scripts/update_schemas to update"),
            file.path(dest, "README.md"))
-file.create(file.path(dest, "__init__.py"))
```
Comment (richfitz) [inline scripts/update_schemas:26]: unrelated change, this file had been manually deleted before

**Review by r-ash — APPROVED** [review]
